grudge/array_context.py

```diff
@@ -241,6 +241,13 @@ def _dag_to_compiled_func(self, dict_of_named_arrays,
 
         distributed_partition = pt.find_distributed_partition(dict_of_named_arrays)
 
+        if __debug__:
+            # pylint-ignore-reason:
+            # '_BasePytatoArrayContext' has no 'mpi_communicator' member
+            pt.verify_distributed_partition(
+                self.actx.mpi_communicator,  # pylint: disable=no-member
+                distributed_partition)
```

Comment:
Is this guaranteed to terminate cleanly or is there a chance we might need a `self.actx.mpi_communicator.abort()` if a failure was seen?

Reply:
IMO, this might better be addressed by using `python -m mpi4py` in the MPI command line for these runs. This will automatically abort/kill other ranks if one goes down with an exception.

See, e.g. https://github.com/inducer/grudge/pull/295

Reply:
Thanks, yep it does get resolved if we do `python -m mpi4py ...`. If that's the only way one is supposed to run grudge apps, feel free to close this.

Reply:
It's not *the only* way, but it's a good way to avoid hangs. Realistically, it's something every user of mpi4py-dependent code should know. Mirgecom does some hand-holding along these lines:

https://github.com/illinois-ceesd/mirgecom/blob/0d8635eb16717d62d83a0e62ddcdb0fb7866c4d7/mirgecom/mpi.py#L109-L159

But I find that too heavy-handed for pytato or grudge.